node/core/dispute-coordinator/src/lib.rs

```diff
@@ -323,12 +341,22 @@ where
 				return Ok(())
 			}
 			FromOverseer::Signal(OverseerSignal::ActiveLeaves(update)) => {
-				handle_new_activations(
-					ctx,
-					&mut overlay_db,
-					&mut state,
-					update.activated.into_iter().map(|a| a.hash),
-				).await?
+				let leaves = update.activated.into_iter().map(|a| a.hash);
+				if !state.recovery_state.complete() {
+					handle_leaf(
```

Comment:
nit: rather than have `handle_leaf` also call `handle_new_activations`, I think it'd be more clear to explicitly invoke it here and rename `handle_leaf` to `handle_startup`

Reply:
Why didn't I think of that!